models/basgra/R/run_BASGRA.R

```diff
@@ -0,0 +1,268 @@
+    "(-)"          , "(-)"          ,                                                # 81:82
+    "(-)"          , "(-)"          , "(-)"           ,                              # 83:85
+    "(d-1)"        , "(mm d-1)"     , "(g N m-2 d-1)" , "(g N m-2 d-1)",             # 86:89
+    "(g N m-2 d-1)", "(g N m-2 d-1)", "(g N m-2 d-1)" , "(g N m-2 d-1)",             # 90:93
+    "(g N m-2 d-1)", "(g N m-2 d-1)",                                                # 94:95
+    "(g N m-2)"    , "(g N g-1 C)"  ,                                                # 96:97
+    "(g N m-2)"    , "(g N g-1 C)"  ,                                                # 98:99
+    "(d d-1)"                                                                        # 100
+  )
+  
+  NOUT <- as.integer( length(outputNames) )
+  
+  
+  ############################# SITE CONDITIONS  ########################
+  # this part corresponds to  initialise_BASGRA_***.R functions  
+  
+  year_start  <- as.integer(start_year)
+  doy_start   <- as.integer(lubridate::yday(start_date))
+  NDAYS       <- as.integer(sum(PEcAn.utils::days_in_year(start_year:end_year))) # could be partial years, change later
+
+  matrix_weather <- mini_met2model_BASGRA(run_met, start_date, start_year, end_date, end_year)
+  
+  calendar_fert     <- matrix( 0, nrow=100, ncol=3 )
+  calendar_Ndep     <- matrix( 0, nrow=100, ncol=3 )
+  calendar_Ndep[1,] <- c(1900,  1,0)
+  calendar_Ndep[2,] <- c(2100, 366, 0)
+  days_harvest      <- matrix( as.integer(-1), nrow=100, ncol=2 )
+  
+  # hardcoding these for now, should be able to modify later on
+  calendar_fert[1,] <- c( 2000, 115, 140*1000/ 10000      ) # 140 kg N ha-1 applied on day 115
+  calendar_fert[2,] <- c( 2000, 150,  80*1000/ 10000      ) #  80 kg N ha-1 applied on day 150
+  #    calendar_fert[3,] <- c( 2001, 123, 0*1000/ 10000      ) # 0 kg N ha-1 applied on day 123
+  calendar_Ndep[1,] <- c( 1900,   1,  2*1000/(10000*365) ) #  2 kg N ha-1 y-1 N-deposition in 1900
+  calendar_Ndep[2,] <- c( 1980, 366, 20*1000/(10000*365) ) # 20 kg N ha-1 y-1 N-deposition in 1980
+  calendar_Ndep[3,] <- c( 2100, 366, 20*1000/(10000*365) ) # 20 kg N ha-1 y-1 N-deposition in 2100
+  days_harvest [1,] <- c( 2000, 150 )
+  days_harvest [2,] <- c( 2000, 216 )
+  
+  days_harvest      <- as.integer(days_harvest)
+  
+  # run  model
+  output <- .Fortran('BASGRA',
+                     run_params,
+                     matrix_weather,
+                     calendar_fert,
+                     calendar_Ndep,
+                     days_harvest,
+                     NDAYS,
+                     NOUT,
+                     matrix(0, NDAYS, NOUT))[[8]]
+
+  
+  ############################# WRITE OUTPUTS ###########################
+  # writing model outputs already in standard format
+  
+  # only LAI and CropYield for now
+  
+    
+  years <- seq(start_year, end_year)
+  for (y in years) {
+    
+    thisyear <- output[, which(outputNames == "year")] == y
```

Comment:
FWIW, pretty sure you don't need the `which` here -- boolean subsetting should do exactly the same thing. No need to change -- just noting to save you typing in the future :wink: